voxblox/src/simulation/simulation_world.cc

```diff
@@ -95,4 +95,69 @@ void SimulationWorld::getPointcloudFromViewpoint(
   }
 }
 
+void SimulationWorld::getNoisyPointcloudFromViewpoint(
+    const Point& view_origin, const Point& view_direction,
+    const Eigen::Vector2i& camera_res, FloatingPoint fov_h_rad,
+    FloatingPoint max_dist, FloatingPoint noise_sigma, Pointcloud* ptcloud,
+    Colors* colors) {
+  // Focal length based on fov.
+  FloatingPoint focal_length = camera_res.x() / (2 * tan(fov_h_rad / 2.0));
+
+  // Calculate transformation between nominal camera view direction and our
+  // view direction. Nominal view is positive x direction.
+  Point nominal_view_direction(1.0, 0.0, 0.0);
+  Eigen::Quaternion<FloatingPoint> ray_rotation =
+      Eigen::Quaternion<FloatingPoint>::FromTwoVectors(nominal_view_direction,
+                                                       view_direction);
+
+  // Now actually iterate over all pixels.
+  for (int u = -camera_res.x() / 2; u < camera_res.x() / 2; ++u) {
```

Comment:
nit: precompute the boundaries
Are you sure this covers the image range correctly? 
With these boundaries it does the following if I'm not mistaken:
```
5x5 image: u,v in [-2, 1] -> should probably be in [-2, 2]
4x4 image: u,v in [-2, 1]
```